llmfoundry/data/text_data.py

```diff
@@ -197,10 +210,18 @@ def _read_binary_tokenized_sample(
         self,
         sample: Dict[str, Any],
     ) -> torch.Tensor:
-        return torch.from_numpy(
-            np.frombuffer(sample['tokens'],
-                          dtype=np.int64)[:self.max_seq_len].copy(),
-        )
+        # Modeling code still expects int64 tensors.
+        if isinstance(sample['tokens'], np.ndarray):
+            return torch.from_numpy(
+                sample['tokens'][:self.max_seq_len].copy(),
+            ).to(torch.int64)
```

Comment:
jw, was this just a typing thing? or did it actually error?

Reply:
it actually errors because torch's CrossEntropyLoss expects labels to have `int64` (aka `LongTensor`) type. And that's why our tests also check that, code casts to LongTensor, etc

Reply:
ah got it